src/main/java/com/exadel/discountwebapp/vendor/mapper/VendorMapper.java

```diff
@@ -25,7 +26,7 @@ public VendorMapper(LocationService locationService) {
 
     public VendorResponseVO toVO(Vendor vendor) {
         VendorResponseVO response = modelMapper.map(vendor, VendorResponseVO.class);
-        response.setLocationId(vendor.getLocation().getId());
+        response.setLocationResponseVO(modelMapper.map(vendor.getLocation(), LocationResponseVO.class));
```

Comment:
probably LocationMapper can be injected and reused here to map Location -> LocationResponseVO

Reply:
+